api/types/database.go

```diff
@@ -215,22 +215,43 @@ func (d *DatabaseV3) SetURI(uri string) {
 
 // GetCA returns the database CA certificate.
 func (d *DatabaseV3) GetCA() string {
+	if d.Status.CACert != "" {
+		return d.Status.CACert
+	}
 	return d.Spec.CACert
 }
 
-// SetCA sets the database CA certificate.
-func (d *DatabaseV3) SetCA(bytes string) {
-	d.Spec.CACert = bytes
+// SetStatusCA sets the database CA certificate in the status field.
+func (d *DatabaseV3) SetStatusCA(ca string) {
+	d.Status.CACert = ca
+}
+
+// IsEmpty returns true if AWS metadata is empty.
+func (a AWS) IsEmpty() bool {
+	return a.AccountID == "" && a.Region == "" && a.RDS.IsEmpty() && a.Redshift.IsEmpty()
```

Comment:
nit: wonder if cmp.Equal(a, AWS{}) will work here.

Reply:
Yeah, reason I didn't use it is to not introduce additional deps to the api module but it looks like we already depend on it anyway. I'll update, it'll be much cleaner.